src/state/transactions/transactionUpdater.tsx

```diff
@@ -6,57 +6,35 @@ import values from 'lodash/values';
 import useBuildTransactionMessage from '@hooks/useBuildTransactionMessage';
 import useBuildRejectedTransactionMessage from '@hooks/useBuildRejectedTransactionMessage';
 import Zoom from '@mui/material/Zoom';
-import { useBlockNumber, useGetBlockNumber } from '@state/block-number/hooks';
-import { updateBlockNumber } from '@state/block-number/actions';
+import { useGetBlockNumber } from '@state/block-number/hooks';
 import EtherscanLink from '@common/components/view-on-etherscan';
-import { TransactionDetails, TransactionReceipt, TransactionTypes } from '@types';
+import { TransactionDetails, TransactionTypes } from '@types';
 import { setInitialized } from '@state/initializer/actions';
 import useTransactionService from '@hooks/useTransactionService';
 import useWalletService from '@hooks/useWalletService';
 import useSafeService from '@hooks/useSafeService';
 import usePositionService from '@hooks/usePositionService';
 import { updatePosition } from '@state/position-details/actions';
 import useLoadedAsSafeApp from '@hooks/useLoadedAsSafeApp';
-import useCurrentNetwork from '@hooks/useCurrentNetwork';
+import useInterval from '@hooks/useInterval';
 import { usePendingTransactions } from './hooks';
-import { checkedTransaction, finalizeTransaction, removeTransaction, transactionFailed } from './actions';
+import {
+  checkedTransaction,
+  checkedTransactionExist,
+  finalizeTransaction,
+  removeTransaction,
+  setTransactionsChecking,
+  transactionFailed,
+} from './actions';
 import { useAppDispatch, useAppSelector } from '../hooks';
 
-interface TxInterface {
-  addedTime: number;
-  receipt?: TransactionReceipt;
-  lastCheckedBlockNumber?: number;
-}
-
-export function shouldCheck(lastBlockNumber: number, tx: TxInterface): boolean {
-  if (tx.receipt) return false;
-  if (!tx.lastCheckedBlockNumber) return true;
-  const blocksSinceCheck = lastBlockNumber - tx.lastCheckedBlockNumber;
-  if (blocksSinceCheck < 1) return false;
```

Comment:
This was the reason why it was not updating the txs, right? The changing network problems.

Reply:
yeah